Source/Core/UICommon/AutoUpdate.cpp

```diff
@@ -232,10 +244,23 @@ void AutoUpdateChecker::TriggerUpdate(const AutoUpdateChecker::NewVersionInforma
   updater_flags["log-file"] = File::GetUserPath(D_LOGS_IDX) + UPDATER_LOG_FILE;
 
   if (restart_mode == RestartMode::RESTART_AFTER_UPDATE)
+  {
     updater_flags["binary-to-restart"] = File::GetExePath();
+  }
 
   // Copy the updater so it can update itself if needed.
   std::string updater_path = File::GetExeDirectory() + DIR_SEP + UPDATER_FILENAME;
+
+#ifdef __APPLE__
+  // On universal builds, the updater app is embedded within the main bundle.
+  std::string embedded_updater_path =
+      File::GetBundleDirectory() + "/Contents/Helpers/" + UPDATER_FILENAME;
+  if (File::Exists(embedded_updater_path))
+  {
+    updater_path = embedded_updater_path;
+  }
```

Comment:
const

Is this just a safeguard for when the transition happens, in case something gets merged in the wrong order or whatever? I can't think of another situation where the embedded updater wouldn't be there.

Reply:
I was thinking maybe someone would distribute an Intel-only or an ARM-only bundle in a fork, and the updater is only embedded when building a universal bundle (since the operation is done in the python script).

But now that I think more about it, I highly doubt any fork uses the Dolphin auto-update system, so yeah, this is a pointless check... Will remove.